### PR TITLE
Wednesday project

### PR DESCRIPTION
**Pull request by MarcoAurelioAntonio**
Create a table named owners with the following columns:
id: integer (set it as autoincremented PRIMARY KEY)
full_name: string
age: integer
Create a table named species with the following columns:
id: integer (set it as autoincremented PRIMARY KEY)
name: string
Modify animals table:
Make sure that id is set as autoincremented PRIMARY KEY
Remove column species
Add column species_id which is a foreign key referencing species table
Add column owner_id which is a foreign key referencing the owners table
Remember all this goes in schema.sql file.
Insert the following data into the owners table:
Sam Smith 34 years old.
Jennifer Orwell 19 years old.
Bob 45 years old.
Melody Pond 77 years old.
Dean Winchester 14 years old.
Jodie Whittaker 38 years old.
Insert the following data into the species table:
Pokemon
Digimon
Modify your inserted animals so it includes the species_id value:
If the name ends in "mon" it will be Digimon
All other animals are Pokemon
Modify your inserted animals to include owner information (owner_id):
Sam Smith owns Agumon.
Jennifer Orwell owns Gabumon and Pikachu.
Bob owns Devimon and Plantmon.
Melody Pond owns Charmander, Squirtle, and Blossom.
Dean Winchester owns Angemon and Boarmon.
Remember these insertions and modifications happen in data.sql file.
Write queries (using JOIN) to answer the following questions:
What animals belong to Melody Pond?
List of all animals that are pokemon (their type is Pokemon).
List all owners and their animals, remember to include those that don't own any animal.
How many animals are there per species?
List all Digimon owned by Jennifer Orwell.
List all animals owned by Dean Winchester that haven't tried to escape.
Who owns the most animals?
Remember all these should be written in queries.sql file.
Take a screenshot of the results of your queries.
Remember to add the screenshot with the results of your queries to your Pull Request description.

image

image

image

image

image

image

image

image

image

image

image

image

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -9,3 +9,29 @@ CREATE TABLE animals (
...
+CREATE TABLE animals (
+  id BIGSERIAL NOT NULL PRIMARY KEY,
+  name varchar(100) NOT NULL,
+  date_of_birth DATE NOT NULL,
+  escape_attempts INT NOT NULL,
+  neutered BOOLEAN NOT NULL,
+  weight_kg DOUBLE PRECISION NOT NULL
+  species_id INT NULL REFERENCES species(id) ON DELETE CASCADE,
+  owner_id INT NULL REFERENCES owners(id) ON DELETE CASCADE,
+);
```
Comment (cwaku) [inline schema.sql]: - [x] As per the requirements, modify the table and not recreate it with updated columns. I see you did it right according to your screenshot but kindly replace this with the commands you used as shown in your screenshot as the table is already being created on lines 3-11 of your `schema.sql`. it should look like the example code below.

```sql
   ALTER TABLE REMOVE species;
   ALTER TABLE animals ADD species_id int REFERENCES species(id);
   ALTER TABLE animals ADD owner_id int REFERENCES owners(id);
```

File: queries.sql [hunk 2]
```diff
@@ -54,4 +54,19 @@ SELECT COUNT(*) AS "Escapee_Animals_Count" FROM animals WHERE escape_attempts =
...
-SELECT species, AVG(escape_attempts) AS "Average_Escape" FROM animals WHERE date_of_birth BETWEEN DATE '1990-01-01' AND '2000-12-31' GROUP BY species;
\ No newline at end of file
+SELECT species, AVG(escape_attempts) AS "Average_Escape" FROM animals WHERE date_of_birth BETWEEN DATE '1990-01-01' AND '2000-12-31' GROUP BY species;
+
+/* wednesday */
+
+alter table animals drop column species; /*DELETE COLUMN species*/
+alter table animals add column species_id INT NULL REFERENCES species(id) ON DELETE CASCADE; /*ADD COLUMN species_id, which is a foreign key referencing species table*/
+alter table animals add column owner_id INT NULL REFERENCES owners(id) ON DELETE CASCADE; /*ADD COLUMN owner_id, which is a foreign key referencing owners table*/
+
```
Comment (cwaku) [inline queries.sql]: - [x] Everything that creates or changes the tables structure should be saved in the `schema.sql` file. Kindly move this in the respective file.

**Review by cwaku — CHANGES_REQUESTED** [review]
Hi @MarcoAurelioAntonio 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!
There are a few issues 🐛  that you still need to work 🛠️  on to go to the next project ⏭️, but you are almost there! 🏃🏾 

## Highlights 
 - [x] Correct GitHub Flow 👍🏾 
 - [x] Good commit history 🥇 
 - [x] Good PR title/description 💯 
 - [x] Good job including the screenshots 🥇 

## Required Changes ♻️
It's ok to have bugs in your code. Please, see this code review as a chance to learn practical coding skills and new tips 🌟  that can improve your code quality 💯 and help you get a better job 💼.

I will guide 📔 you through fixing 🛠️ these bugs, follow my comments down below and make sure you don't miss any of them to get your project approved ✅

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by alexander16108 — APPROVED** [review]
Hi @MarcoAurelioAntonio , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**